mitodl/open-discussions
# Spam exemptions check and feature flag

Blocked by https://github.com/mitodl/salt-ops/pull/1266

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #3086 

#### What's this PR do?
- Adds a new feature flag (`SPAM_EXEMPTIONS`)
- Adds a new spam email regex setting (`SPAM_EMAIL_EXEMPTIONS`)
- Skips spam checking if the feature flag is enabled and the user's email matches the regex pattern.

#### How should this be manually tested?
- With the feature flag disabled, everything should work the same as before.  Create a post or comment and a new `SpamCheckResult` object should be created for it.
- Enable the feature flag and log in as a user with a `@mit.edu` email address. Create a post or comment, there should not be a `SpamCheckResult` object created for it.
